### PR TITLE
feat: service worker refresh

### PR DESCRIPTION
**Pull request by kwokieee**
## Describe your changes
Issue a prompt to users to refresh the browser when an update in the service worker is detected.

## Issue ticket number and link
#69 

## Screenshots (if appropriate):
image

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Review by chownces — CHANGES_REQUESTED** [review]
The current implementation requires the user to close all the tabs of our web app, or do a force refresh in order to receive the new update. I think this would be a better flow/UX:

https://developer.chrome.com/docs/workbox/handling-service-worker-updates/

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import ScrollToTop from './components/ScrollToTop';
 import { ProgressBarStyle } from './components/ProgressBar';
 import NotistackProvider from './components/NotistackProvider';
 import MotionLazyContainer from './components/animate/MotionLazyContainer';
+import { SnackbarUtilsConfigurator } from './utils/snackbarUtil';
```
Comment (therizhao) [inline frontend/src/App.tsx:12]: You don't need a new snackbar for this. You can just import useSnackbar.

And use enqueueSnackbar.

Can just do a search for to see how it works.

**Review by therizhao — CHANGES_REQUESTED** [review]

File: frontend/src/App.tsx (hunk 1 above)
Comment (kwokieee) [reply frontend/src/App.tsx:12]: useSnackbar requires a SnackbarProvider, which is only rendered in App.tsx. The service worker registration can only be done in index.tsx, hence I had to use this SnackbarUtilsConfigurator to place a component in the tree that will be able to access the SnackbarProvider.

**Review by therizhao — APPROVED** [review]
LGTM.

**Comment by chownces** [comment]
superseded by #107

**Pull request closed without merging**